### PR TITLE
[DOCS] Fix broken links for ES API docs move

### PR DESCRIPTION
**Pull request by jrodewig**
elastic/elasticsearch/pull/44238 moved several Elasticsearch APIs to a REST APIs section.

As a result of that move, some pages were converted to anchored sections. This will fix links to those pages once the above PR is re-applied.

I plan to merge this PR simultaneously with:
- elastic/elasticsearch/pull/44238
- elastic/kibana/pull/41001
- elastic/elasticsearch-hadoop/pull/1317
- elastic/stack-docs/pull/411

### CODE REVIEW

File: docs/reference/ml/apis/put-datafeed.asciidoc [hunk 1]
```diff
@@ -93,7 +93,7 @@ those same roles.
   script fields to the {dfeed}.
   The <<ml-detectorconfig,detector configuration objects>> in a job can contain
   functions that use these script fields. For more information,
-  see {ref}/search-request-script-fields.html[Script Fields].
+  see {ref}/search-request-body.html#search-request-script-fields[Script Fields].
```
Comment (lcawl) [inline docs/reference/ml/apis/put-datafeed.asciidoc]:
```suggestion
  see <<search-request-script-fields,Script fields>>.
```

File: docs/reference/ml/apis/update-datafeed.asciidoc [hunk 2]
```diff
@@ -86,7 +86,7 @@ The following properties can be updated after the {dfeed} is created:
   The <<ml-detectorconfig,detector configuration objects>> in a job can contain
   functions that use these script fields.
   For more information,
-  see {ref}/search-request-script-fields.html[Script Fields].
+  see {ref}/search-request-body.html#search-request-script-fields[Script Fields].
```
Comment (lcawl) [inline docs/reference/ml/apis/update-datafeed.asciidoc]: Not sure why these were external links in the first place, so I'm suggesting we convert them to internal links:
```suggestion
  see <<search-request-script-fields,Script fields>>.
```

File: docs/reference/ml/transforms.asciidoc [hunk 3]
```diff
@@ -151,7 +151,7 @@ function. The `datafeed-test1` {dfeed} defines the script field. It contains a
 script that adds two fields in the document to produce a "total" error count.
 
 The syntax for the `script_fields` property is identical to that used by {es}.
-For more information, see {ref}/search-request-script-fields.html[Script Fields].
+For more information, see {ref}/search-request-body.html#search-request-script-fields[Script Fields].
```
Comment (lcawl) [inline docs/reference/ml/transforms.asciidoc]:
```suggestion
For more information, see <<search-request-script-fields,Script fields>>.
```

**Review by lcawl — APPROVED** [review]
Added optional suggestions, otherwise LGTM

**Comment by jrodewig** [comment]
Thanks @lcawl. I've applied those changes.

**Pull request merged**